helix-term/src/commands.rs

```diff
@@ -5519,6 +5520,32 @@ fn increment_impl(cx: &mut Context, increment_direction: IncrementDirection) {
     }
 }
 
+fn yank_diagnostic(cx: &mut Context) {
+    let (view, doc) = current_ref!(cx.editor);
+    let primary = doc.selection(view.id).primary();
+
+    let default_sep = doc.line_ending.as_str();
+    let diag = doc
+        .diagnostics()
+        .iter()
+        .filter(|d| primary.overlaps(&helix_core::Range::new(d.range.start, d.range.end)))
+        .map(|d| d.message.clone())
+        .fold(String::new(), |mut acc, fragment| {
```

Comment:
I wonder if we should just collect this into a Vec instead: the `+` register will join the values before copying to the clipboard and it could be useful to have each diagnostic be a separate entry in the register (for example you could paste just the first value if there are duplicated diagnostics)

Reply:
I've made the change, but I don't think we currently have a way to paste a value by ordinal count.